### ISSUE TITLE
"Reopen folder using..." should use a more dedicated ux solution, not notifications

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#121805]
Refs: https://github.com/microsoft/vscode/issues/119463

Currently we have mulitple experience to reopen the folder because there is a `Dev Container` file or a `.code-workspace` file. Both of them use notifications in order to make the user aware of this.

Instead of this I propose the following:
* Do not use notificaitons for this (especially painful since they appear on startup)
* We should consider showing this in the empty editor area instead of action hints. Button "Reopen in Container" 
* Consider having this in the welcome view
* We should have something in the explorer. Either a Ttitle area button, or a blue button in the empty area

fyi @misolori @bpasero for ideas

![Screenshot 2021-04-21 at 12 21 42](https://user-images.githubusercontent.com/1926584/115538538-46b81880-a29c-11eb-9868-793a4c8c63c6.png)

**Comment by isidorn** [comment]
Also the dev-containers shows multiple notifications, even if I click on don't show again I get another one straight away.
I think we can improve this flow.

![Screenshot 2021-04-22 at 15 26 41](https://user-images.githubusercontent.com/1926584/115722307-28771900-a37f-11eb-8b6a-2141ea3ea3af.png)

**Comment by sandy081** [comment]
Assigning @bpasero for code-workspace notification.

**Comment by bpasero** [comment]
> We should have something in the explorer. Either a Ttitle area button, or a blue button in the empty area

The explorer is not empty in this case though, we just scan the top level files of the folder for workspace files. I am not sure where to put such a button then?

**Comment by bpasero** [comment]
@isidorn btw I think the 2 notifications do not really have something in common right, should there be 2 issues then?

**Comment by isidorn** [comment]
@bpasero oh they have in common that they are offering to reopen the folder based on some file in the workspace. So I think maybe they can be solved with the same UX.
I can of course open separate issues if that will make things easier

I do understand that the Explorer is not empty, but most of the time there is extra space at the bottom of the explorer and maybe we could render some buttons / banner there. Just an idea.

**Comment by misolori** [comment]
We should look at telemetry data to see how many people use the "Reopen in Container" action from the toast. This is more of a discoverability item where we let the user know that they CAN open it in a container if they'd like. I'd be ok removing the toast and using the remote indicator or command palette but not sure if that'd impact the usability of it. cc @Chuxel @bamurtaugh for feedback within the Containers extension.

**Comment by Chuxel** [comment]
@misolori Personally, this is the main way I use the extension.  Its logically similar to the workspace notification raising awareness to the presence of a code-workspace file.  Is the proposal to remove both? 

In my view, the core issue here is that both of these are things that you discover after cloning a repository and opening it. We'd want a similar solution. The dev container one has the additional disadvantage of not being in the file menu, so the primary way I personally get into one is to first type "code ." or use the normal file menu and then "reopen". Speaking for myself, I never use "Open Folder in Container."  Brigit may have some additional data.

**Comment by bamurtaugh** [comment]
I just spent some time looking through notification data, but I can't seem to discern which notification is which. I could discern if users took an action, which action they took, but I'm not sure what % of users were presented that notification/completed the action out of the total # of notifications. Based on action labels alone, it looks like "Reopen in Container" has a notable lead in clicks over "Don't Show Again" (I want to be conscious of listing exact numbers in a public issue - not sure what's ok to post?).

My primary flow is the same as Chuck's - I open a project containing a dev container (either through the file menu, from the WSL file system, or code .), then reopen in a dev container. I usually use the reopen prompt, but occasionally the command. I essentially never use "Open Folder in Container" either.

When I look at the commands executed from Remote-Containers (since the notification is backed by the same command coming from the Command Palette), the number of folks using Open Folder and Reopen in Container appears about equal (again, not sure how precise I can be in a public issue?).

I was a bit surprised to see the equal distribution of Open Folder and Reopen in Container (since I anticipated more user action would come from Reopen as it also has a prompt), so perhaps user behavior is a bit different than our own.

**Comment by isidorn** [comment]
Thanks for sharing your thoughts. 
@Chuxel there is no concrete proposal to remove anything yet, I was just thinking about this problem and if we could introduce some nicer UX that both the Container Reopen and the Workspace Reopen could use.

From your comments it seems like you LIKE having an easy way to click on a button somewhere and the workspace reopens in the appropriate setting. I think this flow makes perfect sense, I am just arguing that maybe that button should not be in a notification, but maybe the explorer or some other place.

Even we do not figure out a better ux, at the very minimum we should polish the "Do Not Show Again" flow for containers which uses 1 extra notification it seems.

Idea:
We need a concept of badges, which would be like footstep clues that would go away after a while. So we put a badge on the explorer icon, once the user clicks on that we would show a badge on a new title level explorer action to "Reopen in Container", "Reopen in Workspace".
In older VS Code versions we had this badges on the Debug Configure action, when we wanted to user to click there.

**Comment by bamurtaugh** [comment]
I agree we could polish the "Do Not Show Again" flow for containers - it feels a bit counterintuitive to select you don't want anymore notifications, and then are immediately shown another one. One option is for us to make the decision for the user. Looking at data, more users choose "Current Folder" than "Any Folder," so we could default to "Current Folder." But perhaps we could optimize the wording or flow further since this wouldn't necessarily be clear (i.e. if I select "Do Not Show Again" in one folder and see it again in another, I might be confused). 

It could be interesting to run an A/B experiment to see if a badge or notification for Remote-Containers is more engaging (although experiments don't convey user sentiment around notification flow/overload). 

Badges could be useful for something like the Remote-WSL recommender too - help draw user attention to the remote indicator since they may not be aware something was installed/anything changed/they should take an action.

**Comment by isidorn** [comment]
@bamurtaugh I like that point for just choosing the "Current Folder" automatically!
As for the badges, maybe I can bring this up in the next UX sync @misolori

**Comment by bpasero** [comment]
I agree that this should have a UX discussion first before removing notifications.

Btw the one for workspace has a "Do not show again" scoped for the workspace (not globally).

**Comment by isidorn** [comment]
For may there are no plans to change anything here after discussing in the UX meeting.
The only low hanging fruit would be to fix the Don't Show Again flow as mentioned by @bamurtaugh in [this comment](https://github.com/microsoft/vscode/issues/121805#issuecomment-825772152)

For that leaving this one assigned to @chrmarti 
And @bpasero and me can maybe jump on this once we have badges
